akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -109,8 +135,47 @@ private[remote] class Encoder(
 
       override def onPull(): Unit = pull(in)
 
+      /**
+       * External call from ChangeOutboundCompression materialized value
+       */
+      override def changeActorRefCompression(table: CompressionTable[ActorRef]): Future[Done] = {
+        val done = Promise[Done]()
+        try changeActorRefCompressionCb.invoke((table, done)) catch {
+          // in case materialization not completed yet
+          case NonFatal(_) ⇒ done.tryFailure(new ChangeOutboundCompressionFailed)
```

Comment:
This is a really harmless failure, it will be retried on next advertisement or handshake attempt. It will only occur when callback is invoked before preStart. That is highly unlikely to happen since advertisement is not done immediately and handshake involves network roundtrip.

I'll add some comments